compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Upgrade.hs

```diff
@@ -340,19 +340,16 @@ checkDefDataType origin datatype = do
           checkFields origin (Upgrading {..})
       Upgrading { _past = DataVariant _past, _present = DataVariant _present } -> do
           let upgrade = Upgrading{..}
-          (existing, new) <- checkDeleted (\_ -> EUpgradeError (VariantRemovedVariant origin)) (fmap HMS.fromList upgrade)
-          when (not (null new)) $
-              throwWithContextF present $ EUpgradeError (VariantAddedVariant origin)
+          (existing, _new) <- checkDeleted (\_ -> EUpgradeError (VariantRemovedVariant origin)) (fmap HMS.fromList upgrade)
```

Comment:
I think this would allow reordering variant constructors, which iirc we don't want

Reply:
Good catch - will fix

Reply:
(incidentally, we needed a check on both variants and on enums0